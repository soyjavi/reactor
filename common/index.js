import { LAYOUT } from './layout/index';
import SHAPE from './shape';
import THEME from './theme';

export { LAYOUT, SHAPE, THEME };
