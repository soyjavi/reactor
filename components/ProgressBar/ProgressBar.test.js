import React from 'react';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import { ProgressBar } from '..';

const { SPACE } = THEME;

jest.useFakeTimers();

describe('<ProgressBar>', () => {
  it('renders', () => {
    const tree = renderer.create(<ProgressBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {progress}', () => {
    const tree = renderer.create(<ProgressBar progress={0.33} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {color}', () => {
    const tree = renderer.create(<ProgressBar color={THEME.COLOR.ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {trackColor}', () => {
    const tree = renderer.create(<ProgressBar trackColor={THEME.COLOR.ERROR} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {style}', () => {
    const tree = renderer.create(<ProgressBar style={{ borderRadius: SPACE.XS, height: SPACE.M }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
