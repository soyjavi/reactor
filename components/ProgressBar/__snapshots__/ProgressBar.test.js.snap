// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar> renders 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 5,
        "overflow": "hidden",
        "width": "100%",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4066EA",
        "height": "100%",
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {color} 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 5,
        "overflow": "hidden",
        "width": "100%",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f44336",
        "height": "100%",
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {progress} 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 5,
        "overflow": "hidden",
        "width": "100%",
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4066EA",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {style} 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 5,
        "overflow": "hidden",
        "width": "100%",
      },
      Object {
        "borderRadius": 8,
        "height": 16,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4066EA",
        "height": "100%",
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`<ProgressBar> when {trackColor} 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "height": 5,
        "overflow": "hidden",
        "width": "100%",
      },
      undefined,
      Object {
        "backgroundColor": "#f44336",
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4066EA",
        "height": "100%",
        "width": 0,
      }
    }
  />
</View>
`;
